### ISSUE TITLE
Executable for Windows

### ISSUE THREAD

**Issue by petrasvestartas** [coin-or/Ipopt#800]
We have been using lpopt for [compas_cra ](https://blockresearchgroup.github.io/compas_cra/latest/installation.html)package:
`conda create -n cra python=3.10 pyomo=6.4.2 ipopt=3.14.9 compas`

This works for **mac** and **linux**, but not for **windows**, so we also need to install very first version` ipopt=3.11.1`
It was working for multiple years for research, but now completely broke.

Is there any way that you could provide executable for 3.14.9 version too?

**Comment by svigerske** [comment]
There are windows binaries available at https://github.com/coin-or/Ipopt/releases/tag/releases%2F3.14.9.
But I don't know the situation for conda packages.

**Comment by petrasvestartas** [comment]
@svigerske Is there anyone could take a look at the compas_cra package in more detail?
We are bit confused by lpop on windows...

**Comment by svigerske** [comment]
Maybe check with someone from https://compas.dev/#/association ? I don't know anything about this.

**Comment by petrasvestartas** [comment]
We are from compas, we have no clue...
Pyomo just do not work as before with Ipopt.

**Comment by svigerske** [comment]
So what does "do not work" mean in detail? What is going wrong where?

**Comment by petrasvestartas** [comment]
I get this error, first I though that it was executable not found, but now I have a doubt about it, since conda installed it:


```
(cra) C:\brg\2_code\compas_cra>C:/Users/petras/anaconda3/envs/cra/python.exe c:/brg/2_code/compas_cra/tests/test_cra_penalty.py
Aeq: (6, 12)
Aeq: (6, 16)
Afr: (32, 16)
1
1
finished setup... now trying to solve it...
ERROR: Solver (ipopt) returned non-zero return code (3221225501)
ERROR: Solver log: Ipopt 3.11.1: tol=1e-08 constr_viol_tol=1e-07
acceptable_tol=1e-06 acceptable_constr_viol_tol=1e-05

**************************************************************************
**** This program contains Ipopt, a library for large-scale nonlinear
optimization.
 Ipopt is released as open source code under the Eclipse Public License
 (EPL).
         For more information visit http://projects.coin-or.org/Ipopt
**************************************************************************
****

NOTE: You are using Ipopt by default with the MUMPS linear solver.
      Other linear solvers might be more efficient (see Ipopt
      documentation).


This is Ipopt version 3.11.1, running with linear solver mumps.

Number of nonzeros in equality constraint Jacobian...:      112 Number of
nonzeros in inequality constraint Jacobian.:      128 Number of nonzeros
in Lagrangian Hessian.............:       54

Total number of variables............................:       26
                     variables with only lower bounds:       12
                variables with lower and upper bounds:        0
                     variables with only upper bounds:        0
Total number of equality constraints.................:       26 Total
number of inequality constraints...............:       48
        inequality constraints with only lower bounds:        4
   inequality constraints with lower and upper bounds:       12
        inequality constraints with only upper bounds:       32

iter    objective    inf_pr   inf_du lg(mu)  ||d||  lg(rg) alpha_du
alpha_pr  ls
   0 4.0000000e+002 1.00e+000 2.00e+004  -1.0 0.00e+000    -  0.00e+000
   0.00e+000   0
```

**Comment by svigerske** [comment]
Looks like Ipopt just crashed. Can you just try with a current Ipopt version?

**Comment by petrasvestartas** [comment]
By replacing them it also crashes without any message:
```

(cra) C:\brg\2_code\compas_cra>C:/Users/petras/anaconda3/envs/cra/python.exe c:/brg/2_code/compas_cra/tests/test_cra_penalty.py
Aeq:  (6, 12)
Aeq:  (6, 16)
Afr:  (32, 16)
1
1
finished setup... now trying to solve it...
ERROR: Solver (ipopt) returned non-zero return code (3221225781)
Traceback (most recent call last):
  File "c:\brg\2_code\compas_cra\tests\test_cra_penalty.py", line 36, in <module>
    test_cra_penalty()
  File "c:\brg\2_code\compas_cra\tests\test_cra_penalty.py", line 21, in test_cra_penalty
    cra_penalty_solve(assembly, density=1)
  File "C:\Users\petras\anaconda3\envs\cra\lib\site-packages\compas_cra\equilibrium\cra_penalty_pyomo.py", line 146, in cra_penalty_solve
    result = solver.solve(model, tee=verbose)
  File "C:\Users\petras\anaconda3\envs\cra\lib\site-packages\pyomo\opt\base\solvers.py", line 596, in solve
    raise ApplicationError(
pyomo.common.errors.ApplicationError: Solver (ipopt) did not exit normally
```

**Comment by svigerske** [comment]
Maybe you can retrieve the `.nl` file that Pyomo generates and passes on to Ipopt and see whether Ipopt also crashes when run on the .nl file independently of cra and pyomo.

If not (or anyway), check the Windows event log whether it gives some details.

**Comment by petrasvestartas** [comment]
I think I might have a solution based on your initial .exe release version. I will triple check and let you know. You might actually have helped to solve this issue... phew...

**Comment by traversaro** [comment]
For reference, we have an issue in https://github.com/conda-forge/ipopt-feedstock/issues/55 for tracking the packaging of the ipopt executable on Windows in conda-forge.

**Comment by petrasvestartas** [comment]
@traversaro @svigerske 

I did the following:
```bash
conda create -n cra python=3.10 pyomo=6.4.2 ipopt=3.14.9 compas
conda activate cra
pip install compas_assembly compas_cra
```

The above works on mac but not on windows, because conda is not shipping the executables. Then I did the following.
I unzipped the compiled executable from the first link of:  https://github.com/coin-or/Ipopt/releases/tag/releases%2F3.14.9. 
and placed bin folder contents to:
`C:\Users\petras\anaconda3\envs\cra_dev\Library\bin`

All my code works!
This is issue is solved. Thank you  @svigerske !

We will ship these executables separately from pyomo conda distributables, sadly it is not the default behavior. But this workaround works.
